client/src/components/DisplayTable.js

```diff
@@ -0,0 +1,140 @@
+import './DisplayTable.scss';
+import classNames from 'classnames';
+import { util_components } from '../panels/shared.js';
+const {
+  SortDirections,
+} = util_components;
+
+import { TM } from './TextMaker.js';
+
+export class DisplayTable extends React.Component {
+  constructor(props){
+    super();
+    this.state = {
+      sort_by: "label",
+      descending: true,
+      filter_by: undefined,
+      filter: undefined,
+      show_all: false,
+    };
+  }
+
+  header_click(col_name){
+    this.setState({
+      sort_by: col_name,
+      descending: (
+        this.state.sort_by === col_name ?
+          !this.state.descending :
+          true
+      ),
+    });
+  }
+
+  render(){
+    const {
+      column_keys,
+      data,
+      table_name,
+      label_col_header,
+      sort_keys,
+      table_data_headers,
+    } = this.props;
+    const {
+      sort_by,
+      descending,
+    } = this.state;
+  
+
+    const sorted_data = _.chain(data)
+      .sortBy(row => _.has(row["sort_keys"],sort_by) ? row["sort_keys"][sort_by] : row["label"]) // for status the sort_key could be 0
```

Comment:
What's the comment here about?

Reply:
There's a way of shortcutting that line that won't work because the sort key might be zero which resolves as falsey, that comment is to warn someone who might be tempted to simplify things

Reply:
Yeah good point, I will keep the comment in there but make it more general, maybe something like "careful with sorting, sort keys could be zero/falsey"